app/scripts/controllers/login.js

```diff
@@ -172,4 +209,17 @@ export function LoginCtrl ($scope, login, $window, $location, $timeout, $rootSco
       /* ------------------Set sessionID------------------ */
     }
   }
+
+  $scope.signInWithKeyCloak = function () {
+    if (config.ssoEnabled) {
+      keycloak.keycloakInstance.init({
+        onLoad: 'login-required',
+        // Must match to the configured value in keycloak
+        redirectUri: $window.location.origin,
+        checkLoginIframe: false
+      })
+    } else {
+      Alerting.AlertAddMsg('login', 'danger', 'Sign-in with Keycloak is disabled. Please contact the server administrator.')
+    }
```

Comment:
The logic in the else statement here will never be invoked because the function `signInWithKeyCloak` is only ever run when the `config.ssoenabled` is true. The button `Sign-In with keycloak` is in a div that has the ng-show directive which depends on `config.ssoenabled`.

Reply:
yes I added that to double make sure this will not be triggered even if there are changes in the views, 
if you think it is not necessary I can remove it

Reply:
i dont think its necessary. Its code that will never be run. Lets remove it

Reply:
Cool done :)